tskv/src/compaction/flush.rs

```diff
@@ -435,6 +277,156 @@ pub async fn run_flush_memtable_job(
     Ok(())
 }
 
+struct WriterWrapper {
+    ts_family_id: TseriesFamilyId,
+    max_level_ts: Timestamp,
+    max_data_block_size: usize,
+
+    /// Buffers of level-0 and level-1 data blocks:
+    ///
+    /// Each variant of DataBlock will be insert to a hard-coded index of buffers:
+    /// `[ [ Float, Integer, Unsigned, Boolean, Bytes ]; 2 ]`
+    buffers: [[DataBlock; 5]; 2],
+    /// Pointer to leve-0 and level-1 TSM writers.
+    writers: [Option<TsmWriter>; 2],
+}
+
+impl WriterWrapper {
+    pub fn new(
+        ts_family_id: TseriesFamilyId,
+        max_level_ts: Timestamp,
+        max_data_block_size: usize,
+    ) -> Self {
+        let data_block_buffers = [
+            DataBlock::new(0, ValueType::Float),
+            DataBlock::new(0, ValueType::Integer),
+            DataBlock::new(0, ValueType::Unsigned),
+            DataBlock::new(0, ValueType::Boolean),
+            DataBlock::new(0, ValueType::String),
+        ];
+        Self {
+            ts_family_id,
+            max_level_ts,
+            max_data_block_size,
+
+            buffers: [data_block_buffers.clone(), data_block_buffers],
+            writers: [None, None],
+        }
+    }
+
+    pub async fn write_field(
+        &mut self,
+        field_id: FieldId,
+        values: &[(Timestamp, FieldVal)],
+        value_type: &ValueType,
+        encoding: DataBlockEncoding,
+        flush_task: &FlushTask,
+    ) -> Result<()> {
+        if values.is_empty() {
+            return Ok(());
+        }
+        let buf_idx = match value_type {
+            ValueType::Float => 0,
+            ValueType::Integer => 1,
+            ValueType::Unsigned => 2,
+            ValueType::Boolean => 3,
+            ValueType::String => 4,
+            ValueType::Unknown => {
+                error!("Flush: Unknown value type for field: {}", field_id);
+                return Ok(());
+            }
+        };
+
+        for (ts, val) in values {
+            let level_idx = if *ts > self.max_level_ts {
```

Comment:
Vector `values` is sorted, so we can make two slices by a binary-search using `max_level_ts` and then iterate them.

Reply:
TODO